Marlin/Configuration_adv.h

```diff
@@ -833,6 +832,106 @@
 #define I2C_SLAVE_ADDRESS  0 // Set a value from 8 to 127 to act as a slave
 
 /**
+ *  
+ *  Spindle control
+ * 
+ *  The M3, M4 & M5 g-code commands are used to turn the spindle on and off & to set the
+ *  spindle speed.  In this implementation the M4 command is the same as the M3 (assumes
+ *  the spindle rotates in one direction only).
+ * 
+ *  SuperPid is a router/spindle speed controller used in the CNC milling community.
+ *  Marlin can be used to turn the spindle on and off.  It can also be used to set
+ *  the spindle speed from 5,000 to 30,000 RPM.  
+ *
+ *  You'll need to select a 3.3V - 5V signal for the ON/OFF function and optionally
+ *  choose a 0-5V PWM pin for the speed control.
+ *
+ *  uncomment the following to enable the ON/OFF function
+ */
+ 
+//#define SPINDLE_ENABLE
+ 
+ //  uncomment the following to enable speed control
+//#define SPINDLE_SPEED  
+
+ // If SPINDLE_SPEED is enabled but SPINDLE_ENABLE isn't then SPINDLE_SPEED is ignored.
+ 
+#define SPINDLE_ENABLE_INVERT  false   // set to "true" if  the spindle on/off function is reversed
+#define SPINDLE_SPEED_INVERT  true   // set to "true" if the spindle speeds up when you want it to go slower
+#define SPINDLE_POWER_UP_DELAY  5    // delay in seconds to allow the spindle to come up to speed
+#define SPINDLE_POWER_DOWN_DELAY  5    // delay in seconds to allow the spindle to stop
+
+/**
+ *  The M3 & M4 commands use the following equation to convert PWM duty cycle to spindle speed 
+ * 
+ *  RPM = PWM duty cycle * RPM_slope  +  RPM_intercept
+ *    where PWM duty cycle varies from 0 to 100
+ *
+ *  set the following for your controller (ALL MUST BE SET)
+ */
+ 
+#define RPM_SLOPE 302.00    // SuperPID & AzteegX3pro controller pin 5
+#define RPM_INTERCEPT 0     // SuperPID & AzteegX3pro controller pin 5
+#define RPM_MIN  5000       // SuperPID
+#define RPM_MAX 30000       // SuperPID
+ 
+/**
+ *
+ *  In the pins.h file for your board you'll need to add the following:
+ *   #define SPINDLE_ENABLE_PIN xx     // xx is the digital pin number
+ *   #define SPINDLE_SPEED_PIN yy      // yy is the digital pin number
```

Comment:
Under the new paradigm we're pre-defining the recommended pins in all the pins files that we can figure out. So if you have some idea of which pins _should_ be used, please add them to the pins files. You should at least add them to the `pins_RAMPS.h` file because this applies to most of the boards we support.